src/main/java/org/folio/orders/utils/PermissionsUtil.java

```diff
@@ -0,0 +1,46 @@
+package org.folio.orders.utils;
+
+import io.vertx.core.json.JsonArray;
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.rest.core.models.RequestContext;
+
+import java.util.List;
+import java.util.Set;
+
+public class PermissionsUtil {
+
+  private static final String PERMISSION_ORDER_APPROVE = "orders.item.approve";
+  private static final String PERMISSION_ORDER_UNOPEN = "orders.item.unopen";
+  private static final String PERMISSION_ORDER_REOPEN = "orders.item.reopen";
+  private static final String EMPTY_ARRAY = "[]";
+
+  public static final String OKAPI_HEADER_PERMISSIONS = "X-Okapi-Permissions";
+
+  private PermissionsUtil() {}
+
+  public static boolean isUserDoesNotHaveDesiredPermission(AcqDesiredPermissions acqPerm, RequestContext requestContext) {
+    return !getProvidedPermissions(requestContext).contains(acqPerm.getPermission());
+  }
+
+  public static boolean isManagePermissionRequired(Set<String> newAcqUnits, Set<String> acqUnitsFromStorage) {
+    return !CollectionUtils.isEqualCollection(newAcqUnits, acqUnitsFromStorage);
+  }
+
+  private static List<String> getProvidedPermissions(RequestContext requestContext) {
+    return new JsonArray(requestContext.getHeaders().getOrDefault(OKAPI_HEADER_PERMISSIONS, EMPTY_ARRAY)).stream()
+      .map(Object::toString)
+      .toList();
+  }
+
+  public static boolean isUserNotHaveApprovePermission(RequestContext requestContext) {
+    return !getProvidedPermissions(requestContext).contains(PERMISSION_ORDER_APPROVE);
+  }
+
+  public static boolean isUserNotHaveUnopenPermission(RequestContext requestContext) {
+    return !getProvidedPermissions(requestContext).contains(PERMISSION_ORDER_UNOPEN);
+  }
+
+  public static boolean isUserNotHaveReopenPermission(RequestContext requestContext) {
+    return !getProvidedPermissions(requestContext).contains(PERMISSION_ORDER_REOPEN);
+  }
```

Comment:
these lines can be improved
for naming -> can be change to hasDesiredPermission, hasReopenPermission, ....
for method extraction !getProvidedPermissions(requestContext).contains(acqPerm.getPermission()); can be extracted to hasPermission method
instead of List, probably we can use Set to be more efficient

Reply:
will be addressed in the subsequent PR